contracts/core/OTCWrapper.sol

```diff
@@ -0,0 +1,690 @@
+        address otoken = IOtokenFactoryWrapperInterface(addressbook.getOtokenFactory()).createOtoken(
+            order.underlying,
+            USDC,
+            _collateralAsset,
+            order.strikePrice,
+            order.expiry,
+            order.isPut
+        );
+
+        // scales by 1e8 for division with oracle price
+        // scales by 1e2 to increase from 6 decimals (notional) to 8 decimals (otoken)
+        uint256 mintAmount = order.notional.mul(1e10).div(oracle.getPrice(order.underlying));
+
+        actions[2] = UtilsWrapperInterface.ActionArgs(
+            UtilsWrapperInterface.ActionType.MintShortOption,
+            address(this), // owner
+            address(this), // address to transfer to
+            otoken, // option address
+            vaultID, // vaultId
+            mintAmount, // amount
+            0, // index
+            "" // data
+        );
+
+        // execute actions
+        controller.operate(actions);
+
+        // transfer otokens to user
+        IERC20(otoken).safeTransfer(order.buyer, mintAmount);
+
+        // order accounting
+        orders[_orderID].premium = _premium;
+        orders[_orderID].collateral = _collateralAsset;
+        orders[_orderID].seller = msg.sender;
+        orders[_orderID].vaultID = vaultID;
+        orders[_orderID].oToken = otoken;
+        orderStatus[_orderID] = OrderStatus.Succeeded;
+
+        emit OrderExecuted(_orderID, _collateralAsset, _premium, msg.sender, vaultID, otoken, totalInitialMargin);
+    }
+
+    /**
+     * @notice settles the vault after expiry
+     * @dev can only be called by the market maker who is the order seller
+     * @param _orderID order id
+     */
+    function settleVault(uint256 _orderID) external nonReentrant {
+        require(_orderID <= latestOrder, "OTCWrapper: inexistent order");
+
+        Order memory order = orders[_orderID];
+
+        require(order.seller == msg.sender, "OTCWrapper: sender is not the order seller");
+
+        UtilsWrapperInterface.ActionArgs[] memory actions = new UtilsWrapperInterface.ActionArgs[](1);
+
+        actions[0] = UtilsWrapperInterface.ActionArgs(
+            UtilsWrapperInterface.ActionType.SettleVault,
+            address(this), // owner
+            msg.sender, // address to transfer to
+            address(0), // not used
+            order.vaultID, // vaultId
+            0, // not used
+            0, // not used
+            "" // not used
+        );
+
+        // execute actions
+        controller.operate(actions);
+
+        // clear maintenance margin
+        marginRequirements.clearMaintenanceMargin(msg.sender, order.vaultID);
+
+        emit VaultSettled(_orderID);
+    }
+
+    /************************************************
+     *  MISCELLANEOUS
+     ***********************************************/
+
+    /**
+     * @dev overrides _msgSender() related to ERC2771
+     */
+    function _msgSender()
```

Comment:
do we need to override these here? they are already in ERC2771ContextUpgradeable

Reply:
yes, otherwise it won't compile